src/formElementPlugins/ApplicationViewWrapper.tsx

```diff
@@ -100,19 +102,19 @@ const ApplicationViewWrapper: React.FC<ApplicationViewWrapperProps> = (props) =>
     return newValidationState
   }
 
-  const onSave = async (jsonValue: ResponseFull) => {
-    if (!jsonValue?.customValidation) {
+  const onSave = async (response: ResponseFull) => {
+    if (!response?.customValidation) {
       // Validate and Save response -- generic
-      const validationResult: ValidationState = await onUpdate(jsonValue?.text)
-      if (jsonValue?.text !== undefined)
+      const validationResult: ValidationState = await onUpdate(response?.text)
+      if (response?.text !== undefined)
         await onSaveUpdateMethod({
           variables: {
             id: currentResponse?.id as number,
-            value: jsonValue,
+            value: response,
             isValid: validationResult.isValid,
           },
         })
-      if (jsonValue === null)
+      if (response === null || response?.text == undefined)
```

Comment:
```suggestion
      if (response === null || !response?.text)
```

I think it's more descriptive... Or `!!response?.text` in case `text` can also be null

Reply:
We have to be careful with this, it's caught us out before. It's checking _specifically_ for undefined in text, not anything that could be falsy (e.g. `""`).  I think your suggestion wouldn't distinguish between undefined or "".
<img width="243" alt="Screen Shot 2021-06-01 at 12 07 26 PM" src="https://user-images.githubusercontent.com/5456533/120250357-f1771a00-c2d1-11eb-9dcb-04111c481184.png">

Reply:
ok. I was considering that "" wouldn't also be a valid response.

Reply:
Yeah, if you delete your response in a Text input, the value goes back to `""`, not undefined -- it never goes undefined again.